doc/api/buffer.md

```diff
@@ -2622,6 +2623,364 @@ buf.fill(0);
+<tr><td><code>'csgb2312'</code></td></tr>
+<tr><td><code>'csiso58gb231280'</code></td></tr>
+<tr><td><code>'gb2312'</code></td></tr>
+<tr><td><code>'gb_2312'</code></td></tr>
+<tr><td><code>'gb_2312-80'</code></td></tr>
+<tr><td><code>'gbk'</code></td></tr>
+<tr><td><code>'iso-ir-58'</code></td></tr>
+<tr><td><code>'x-gbk'</code></td></tr>
+
+<tr><td><code>'gb18030'</code></td><td><code>'gb18030'</code></td><td>Yes</td></tr>
+
+<tr><td rowspan="5"><code>'big5'</code></td><td><code>'big5'</code></td><td rowspan="5">Yes</td></tr>
+<tr><td><code>'big5-hkscs'</code></td></tr>
+<tr><td><code>'cn-big5'</code></td></tr>
+<tr><td><code>'csbig5'</code></td></tr>
+<tr><td><code>'x-x-big5'</code></td></tr>
+
+<tr><td rowspan="3"><code>'euc-jp'</code></td><td><code>'cseucpkdfmtjapanese'</code></td><td rowspan="3">Yes</td></tr>
+<tr><td><code>'euc-jp'</code></td></tr>
+<tr><td><code>'x-euc-jp'</code></td></tr>
+
+<tr><td rowspan="2"><code>'iso-2022-jp'</code></td><td><code>'csiso2022jp'</code></td><td rowspan="2">Yes</td></tr>
+<tr><td><code>'iso-2022-jp'</code></td></tr>
+
+<tr><td rowspan="8"><code>'shift_jis'</code></td><td><code>'csshiftjis'</code></td><td rowspan="8">Yes</td></tr>
+<tr><td><code>'ms932'</code></td></tr>
+<tr><td><code>'ms_kanji'</code></td></tr>
+<tr><td><code>'shift-jis'</code></td></tr>
+<tr><td><code>'shift_jis'</code></td></tr>
+<tr><td><code>'sjis'</code></td></tr>
+<tr><td><code>'windows-31j'</code></td></tr>
+<tr><td><code>'x-sjis'</code></td></tr>
+
+<tr><td rowspan="10"><code>'euc-kr'</code></td><td><code>'cseuckr'</code></td><td rowspan="10">Yes</td></tr>
+<tr><td><code>'csksc56011987'</code></td></tr>
+<tr><td><code>'euc-kr'</code></td></tr>
+<tr><td><code>'iso-ir-149'</code></td></tr>
+<tr><td><code>'korean'</code></td></tr>
+<tr><td><code>'ks_c_5601-1987'</code></td></tr>
+<tr><td><code>'ks_c_5601-1989'</code></td></tr>
+<tr><td><code>'ksc5601'</code></td></tr>
+<tr><td><code>'ksc_5601'</code></td></tr>
+<tr><td><code>'windows-949'</code></td></tr>
+</table>
+
+*Note*: The `'iso-8859-16'` encoding listed in the [WHATWG Encoding Standard][]
+is not supported.
+
+### new buffer.TextDecoder([encoding[, options]])
+
+* `encoding` {string} Identifies the `encoding` that this `TextDecoder` instance
+  supports. Defaults to `'utf-8'`.
+* `options` {object}
+  * `fatal` {boolean} `true` if decoding failures are fatal. Defaults to
+    `false`.
+  * `ignoreBOM` {boolean} When `true`, the `TextDecoder` will include the byte
+     order mark in the decoded result. When `false`, the byte order mark will
+     be removed from the output. This option is only used when `encoding` is
+     `'utf-8'`, `'utf-16be'` or `'utf-16le'`. Defaults to `false`.
+
+Creates an new `TextDecoder` instance. The `encoding` may specify one of the
+supported encodings or an alias.
+
+### textDecoder.decode([input[, options]])
+
+* `input` {ArrayBuffer|TypedArray} An `ArrayBuffer` or `%TypedArray%` instance
```

Comment:
Also `DataView`. The `%TypedArray%` notation is generally reserved to specs, so do you think it is appropriate to use it here?